portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -56,24 +56,23 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         // Send the JSON as the response
         response.setContentType("application/json");
         Gson gson = new Gson();
-        String json = gson.toJson(allMessages, gson.newJsonWriter(response.writer()));
```

Comment:
```gson.toJson(allMessages, gson.newJsonWriter(response.writer()));```

is all we need; we don't need `response.getWriter().println(...)` since passing in the responsewriter means gson will write the output there. This also is a bit more efficient so that we don't create an intermediate string object.